spec/visual/samples/lustre

```diff
@@ -0,0 +1,244 @@
+
+-------------------------------------------------------------------
+(* Compute the roof motion speed *)
+  
+const period = 0.1;  -- in seconds
+const wheel_girth = 1.4;  -- meter
+const size = 50; -- size of the sliding windows used to smooth values
+const max_roof_speed = 10.0;
+const speed_max = 110.0;                                           --
+const speed_hist= 5.0; 
+
+-- Beyong this value, the sensor can not see any obstacle
+const max_dist_sensor = 200.0;
+
+type roof_state = enum { locked, in_motion };
+
+node roof(Tick, Parked, OnOff, Done: bool) returns (Locked:bool; Roof_Speed:real);
+var 
+  pst, st : roof_state;
+  Tick_on_in_motion : bool when in_motion(st);
+let
+  pst = locked fby st;
+  st =  merge pst  
+         (locked    -> if (OnOff and Parked) when locked(pst) then in_motion else locked)
+         (in_motion -> if Done when in_motion(pst) then locked else in_motion);
+  Locked = (st=locked) ;
+  Tick_on_in_motion = Tick when in_motion(st);
+  Roof_Speed = merge st
+    (locked    -> 0.0 when locked(st))
+    (in_motion -> roof_speed(Tick_on_in_motion));
+tel
+
+type roof_speed_state = enum { wait, fast, slow };

```

Comment:
Could you delete the text that's duplicative of earlier code in the sample? This will help in the future when working on the lexer and ensuring there is adequate coverage in the sample.

Reply:
I'm not sure why you say it's a duplicate. If I remove the definition of roof_speed_state type,
It won't be a working example for sure (afa the lustre compiler is concerned.

Reply:
I'm sorry—again, poorly worded on my part.

The comment was meant as a comment on the visual sample generally, not this specific line. A large portion of the visual sample looks to me to check text that doesn't test anything in the lexer that an earlier line didn't test. Some duplication is fine but it's preferable to have a visual sample that more clearly tests the various rules in the lexer. That way, it's easy to see if there's a test of each rule and where to add lines to the sample over time as bugs and edge cases are discovered.

Reply:
Ok. So it's ok if my example does not compile because some definitions are missing?

Reply:
Yeah, the sample doesn't need to be compilable code. Its purpose is to provide a visual way to check lexing. In your branch, if you run `rackup`, you should start a local server that you can visit to see what the lexed version of your sample will look like.

Reply:
ok, so I've shortened it.

Reply:
It still seemed to have a lot of duplication. I've reorganised it a little and pushed an updated version to your branch. Let me know if it looks good to go on your end.

Reply:
It's fine. Thanks a lot.